builders/build/eth-api/verify-contracts/etherscan-plugins.md

```diff
@@ -137,4 +138,79 @@ If the contract verification was successful, in your terminal, you should see **
 
 ![Successful verification using truffle-verify-plugin](/images/builders/build/eth-api/verify-contracts/etherscan-plugins/plugins-4.png)
 
-For further information on the plugin, please refer to the [README.md file](https://github.com/rkalis/truffle-plugin-verify#readme){target=_blank} of the `truffle-plugin-verify` GitHub repository.
\ No newline at end of file
+For further information on the plugin, please refer to the [README.md file](https://github.com/rkalis/truffle-plugin-verify#readme){target=_blank} of the `truffle-plugin-verify` GitHub repository.
+
+## Using Foundry to Verify {: #using-foundry-to-verify }
+
+The example in this section of the guide will use the `MyToken.sol` contract that was created in the [Using Foundry to Deploy to Moonbeam](/builders/build/eth-api/dev-env/foundry/){target=_blank} guide.
+
+In addition to the Foundry project, you will need a [Moonbeam Moonscan](https://moonscan.io/){target=_blank} API key. This API key can be used for both the Moonbeam and Moonbase Alpha networks. If you want to verify a contract on Moonriver, you'll need a [Moonriver Moonscan](https://moonriver.moonscan.io/){target=_blank} API key. 
+
+If you have already deployed the example contract, you can verify it with the `verify-contract` command. Before you can verify the contract, you will need to ABI-encode the constructor arguments. To do so for the example contract, you can run the following command:
+
+```
+cast abi-encode "constructor(uint256)" 100
+```
+
+The result should be `0x0000000000000000000000000000000000000000000000000000000000000064`. You can then verify the contract using the following command:
+
+=== "Moonbeam"
+    ```
+    forge verify-contract --chain-id {{ networks.moonbeam.chain_id }} \
+    YOUR_CONTRACT_ADDRESS \
+    --constructor-args 0x0000000000000000000000000000000000000000000000000000000000000064 \
+    src/MyToken.sol:MyToken \
+    YOUR_MOONSCAN_API_KEY
+    ```
+
+=== "Moonriver"
+    ```
+    forge verify-contract --chain-id {{ networks.moonriver.chain_id }} \
+    YOUR_CONTRACT_ADDRESS \
+    --constructor-args 0x0000000000000000000000000000000000000000000000000000000000000064 \
+    src/MyToken.sol:MyToken \
+    YOUR_MOONSCAN_API_KEY
+    ```
+
+=== "Moonbase Alpha"
+    ```
+    forge verify-contract --chain-id {{ networks.moonbase.chain_id }} \
+    YOUR_CONTRACT_ADDRESS \
+    --constructor-args 0x0000000000000000000000000000000000000000000000000000000000000064 \
+    src/MyToken.sol:MyToken \
+    YOUR_MOONSCAN_API_KEY
+    ```
+
+![Foundry Verify](/images/builders/build/eth-api/verify-contracts/etherscan-plugins/plugins-5.png)
```

Comment:
The reason that the verify contract call was failing before was because the constructor args weren't passed in - can you please update this screenshot? Then this PR will be good to go 🚀

Reply:
Screenshot updated. Thanks for the help 🧠

Reply:
Thank you! 🙌